src/services/tokens.ts

```diff
@@ -49,27 +49,29 @@ export class TokenService extends CommonService {
         const txHash = await this.finP2PContract.associateAsset(assetId, tokenAddress);
         await this.finP2PContract.waitForCompletion(txHash);
 
-        return assetCreationResult(txHash, tokenAddress, tokenAddress, this.finP2PContract.finP2PContractAddress);
+        return assetCreationResult(txHash, tokenId, tokenAddress, this.finP2PContract.finP2PContractAddress);
 
       } else {
 
         // We do deploy ERC20 here and then associate it with the FinP2P assetId,
         // in a real-world scenario, the token could already deployed in another tokenization application,
         // so we would just associate the assetId with existing token address
-        let tokenAddress: string;
+        let tokenId, tokenAddress: string;
         switch (this.assetCreationPolicy.type) {
           case 'deploy-new-token':
             tokenAddress = await this.finP2PContract.deployERC20(assetId, assetId,
               this.finP2PContract.finP2PContractAddress);
+            tokenId = tokenAddress;
             break;
           case 'reuse-existing-token':
             tokenAddress = this.assetCreationPolicy.tokenAddress;
+            tokenId = `${getRandomNumber(10000, 100000)}-${tokenAddress}`;
```

Comment:
why between 10_000 and 100_000? just interesting

Reply:
that policy was added specifically for e2e and load testing, it mimics a creation of a new ERC-20 token, by taking the same token again. The pick of the pefix was based on how many assets we may have in e2e :)